kong/plugins/response-transformer/header_transformer.lua

```diff
@@ -0,0 +1,84 @@
+local stringy = require "stringy"
+
+local table_insert = table.insert
+local unpack = unpack
+local type = type
+
+local APPLICATION_JSON = "application/json"
+local CONTENT_TYPE = "content-type"
+local CONTENT_LENGTH = "content-length"
+
+local _M = {}
+
+local function iter(config_array)
+  return function(config_array, i, previous_header_name, previous_header_value)
+    i = i + 1
+    local header_to_test = config_array[i]
+    if header_to_test == nil then -- n + 1
+      return nil
+    end
+    local header_to_test_name, header_to_test_value = unpack(stringy.split(header_to_test, ":"))
+    return i, header_to_test_name, header_to_test_value  
+  end, config_array, 0
+end
+
+local function append_value(current_value, value)
+  local current_value_type = type(current_value)
+ 
+  if current_value_type  == "string" then
+    return {current_value, value}
+  elseif current_value_type  == "table" then
+    table_insert(current_value, value)
+    return current_value  
+  else
+    return {value} 
+  end
+end
+
+local function is_json_body(content_type)
+  if content_type then
+    content_type = stringy.strip(content_type):lower()
+    return stringy.startswith(content_type, APPLICATION_JSON)
+  end
+  return false
+end
```

Comment:
This function should probably be implemented the same way as the one in body_transformer.